File: java/src/main/java/com/sasteval/vuln/easy/XssDirect.java
package com.sasteval.vuln.easy;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;

/**
 * CWE-79: Cross-Site Scripting (Reflected, Direct)
 * VULNERABILITY: User input is reflected directly into HTML output without encoding.
 */
public class XssDirect extends HttpServlet {

    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {

        String name = request.getParameter("name");
        response.setContentType("text/html");

        // VULN: Direct reflection of user input into HTML
        response.getWriter().println("<h1>Hello " + name + "</h1>");
    }
}

File: dotnet/Vuln/Easy/XssDirect.cs
namespace SastEval.Vuln.Easy;

/// <summary>
/// CWE-79: Cross-Site Scripting (Reflected) - User input rendered directly in HTML without encoding.
/// </summary>
public static class XssDirect
{
    public static IResult Handle(HttpContext context)
    {
        var name = context.Request.Query["name"].ToString();

        if (string.IsNullOrEmpty(name))
        {
            return Results.BadRequest("Missing 'name' query parameter");
        }

        // VULNERABLE: User input directly interpolated into HTML response
        return Results.Content($"<h1>Hello {name}</h1>", "text/html");
    }
}
